functions/test/endpoints/PointLog-Tests/Handle.test.ts

```diff
@@ -59,76 +63,197 @@ describe('point_log/handle', ()  => {
+        res.end(function (err, res) {
+            if (err) {
+                done(err)
+            } else {
+                expect(res.status).toBe(413)
+                done()
+            }
+        })
+    })
 
-    // Test if document_id is missing
-    
-    // Test if document_id cannot be found
-    
     // Test approve when currently rejected
-    // it('Approve when rejected', async(done) => {
-    //     const body = {"approve":"true", "approver_id": RHP_ID, "document_id": rejected_log}
-    //     const res: request.Test = factory.post(point_log_func, HANDLE_POINT_PATH, body, RHP_ID)
-    //     res.end(function (err, res) {
-    //         if (err) {
-    //             done(err)
-    //         } else {
-    //             expect(res.status).toBe(200)
-    //             done()
-    //         }
-    //     })
-    // })
+    it('approve when rejected', async(done) => {
+        const body = {"approve":true, "point_log_id":unhandled_log}
+        const res: request.Test = factory.post(point_log_func, HANDLE_POINT_PATH, body, RHP_ID)
+        res.end(function (err, res) {
+            if (err) {
+                done(err)
+            } else {
+                expect(res.status).toBe(201)
+                done()
+            }
+        })
+    })
     
     // Test approve when currently approved
+    it('approve when already approved', async(done) => {
+        const body = {"approve":true, "point_log_id":approved_log}
+        const res: request.Test = factory.post(point_log_func, HANDLE_POINT_PATH, body, RHP_ID)
+        res.end(function (err, res) {
+            if (err) {
+                done(err)
+            } else {
+                expect(res.status).toBe(416)
+                done()
+            }
+        })
+    })
 
     // Test reject when currently approved
-    
+    it('reject when approved', async(done) => {
+        const body = {"approve":false, "point_log_id":approved_log}
+        const res: request.Test = factory.post(point_log_func, HANDLE_POINT_PATH, body, RHP_ID)
+        res.end(function (err, res) {
+            if (err) {
+                done(err)
+            } else {
+                expect(res.status).toBe(201)
+                done()
+            }
+        })
+    })
+
     // Test reject when currently rejected
+    it('reject when already rejected', async(done) => {
+        const approve_body = {"approve":false, "point_log_id":unhandled_log}
+        await factory.post(point_log_func, HANDLE_POINT_PATH, approve_body, RHP_ID)
+        const body = {"approve":false, "point_log_id":unhandled_log}
+        const res: request.Test = factory.post(point_log_func, HANDLE_POINT_PATH, body, RHP_ID)
+        res.end(function (err, res) {
+            if (err) {
+                done(err)
+            } else {
+                expect(res.status).toBe(416)
+                done()
+            }
+        })
+    })
```

Comment:
For each of the tests which should be a success, we want to verify also that the data is correctly changed. After you verify the response code is correct, can you add some checks to confirm that the user's points, user's semester points, and the house's points were correctly updated?

Reply:
You're 100% right. Especially since that add 5 points got left in. Sorry I was being lazy haha ;)